pandas/tests/frame/methods/test_replace.py

```diff
@@ -661,6 +661,20 @@ def test_replace_simple_nested_dict_with_nonexistent_value(self):
         result = df.replace({"col": {-1: "-", 1: "a", 4: "b"}})
         tm.assert_frame_equal(expected, result)
 
+    def test_replace_NA_with_None(self):
```

Comment:
in all of the relevant examples the both the value being replaced and the replacement are NA.  are these the only affected cases?

Reply:
IIUC for a list like to_replace None is treated explicitly at the moment, whereas if using a scalar None, the behavior is different in some cases. My understanding is that users are therefore using a dictionary to get the explicit replacement behavior. To make these consistent, we would need to deprecate this?